src/Discord.Net.Commands/CommandService.cs

```diff
@@ -506,19 +506,66 @@ public async Task<IResult> ExecuteAsync(ICommandContext context, string input, I
             services = services ?? EmptyServiceProvider.Instance;
 
             var searchResult = Search(input);
-            if (!searchResult.IsSuccess)
+
+            //Since ValidateAndGetBestMatch is deterministic on the return type, we can use pattern matching on the type for infering the code flow.
+            var (validationResult, commandMatch) = await ValidateAndGetBestMatch(searchResult, context, services, multiMatchHandling);
+
+            if(validationResult is SearchResult)
```

Comment:
Tho i understand the way you are using this workflow to branch out the execution method, i dont think it is not the most descript way of doing this. I think this should be more user readable to make it easier to maintain. Using an `IResult` to return the method result, with some additional info for branching out the pipeline, might be a better idea.

Reply:
Understood.